modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -98,7 +107,7 @@ public void deleteAssetLibraryUserAccount(
 		}
 
 		Group group = _getGroup(assetLibraryExternalReferenceCode);
-		User user = _userService.getUserByExternalReferenceCode(
+		User user = _userLocalService.getUserByExternalReferenceCode(
 			userAccountExternalReferenceCode, contextCompany.getCompanyId());
 
 		return getAssetLibraryUserAccount(group.getGroupId(), user.getUserId());
```

Comment:
Instead of having one public method call another public method (with the caller not doing any check), can you refactor this so that both public methods do the permission checks first, and then call a private method that assumes permissions are ok?
```
public UserAccount getAssetLibraryUserAccount(...) {
    _checkAssetLibraryAdminOrAssetLibraryMember(assetLibraryId);
    _checkAssetLibraryMember(assetLibraryId, userAccountId);

    return _getAssetLibraryUserAcount(...);
}
```

Reply:
That's better. Done.